ProcessQueue/__init__.py

```diff
@@ -266,6 +266,302 @@ def process_failure(api_url, scan_report_id, headers):
+    value_names = flatten(value_names)
+    print("Unfiltered values",value_names)
+
+    # get all scan reports to be used to filter values by only values that come from active scan reports
+    get_scan_reports = requests.get(
+            url=f"{api_url}scanreports/",
+            headers=headers,
+        )
+    # get active scanreports and map them to fields. Remove any fields in archived reports
+    scanreports =json.loads(get_scan_reports.content.decode("utf-8"))
+    active_reports = [str(item['id']) for item in scanreports if item["hidden"] == False]  
+    # active reports is list of report ids that are not archived
+    
+    # get field ids from values and use to get scan report fields
+    field_ids = set([item['scan_report_field'] for item in value_names])
+    paginated_field_ids = paginate_chars(field_ids,"")
+    fields = []
+    for Ids in paginated_field_ids:
+        ids_to_get = ",".join(map(str, Ids))
+        
+        get_value_fields = requests.get(
+            url=f"{api_url}scanreportfieldsfilter/?id__in={ids_to_get}",
+            headers=headers,
+        )
+        fields.append(json.loads(get_value_fields.content.decode("utf-8")))   
+    fields = flatten(fields)
+    # get table id's from fields and repeat the process
+    table_ids = set([item['scan_report_table'] for item in fields])
+    paginated_table_ids = paginate_chars(table_ids,"")
+    tables = []
+    for Ids in paginated_table_ids:
+        ids_to_get = ",".join(map(str, Ids))
+        
+        get_field_tables = requests.get(
+            url=f"{api_url}scanreporttablesfilter/?id__in={ids_to_get}",
+            headers=headers,
+        ) 
+        tables.append(json.loads(get_field_tables.content.decode("utf-8"))) 
+    tables = flatten(tables)
+    # map value id to active scan report
+    table_id_to_scanreport_map ={str(element.get("id", None)): str(element.get("scan_report", None)) for element in tables}
+    field_id_to_active_scanreport_map ={str(element.get("id", None)): table_id_to_scanreport_map[str(element.get("scan_report_table", None))] for element in fields if  str(table_id_to_scanreport_map[str(element.get("scan_report_table", None))]) in active_reports}
+    value_id_to_active_scanreport_map = {str(element.get("id", None)): field_id_to_active_scanreport_map[str(element.get("scan_report_field", None))] for element in value_names if str(element.get("scan_report_field", None)) in field_id_to_active_scanreport_map}
+    value_names = [item for item in value_names if str(item["id"]) in value_id_to_active_scanreport_map]
+    print("VALUES TO SCAN REPORT",value_id_to_active_scanreport_map)
+    print("FILTERED VALUES",value_names)
+
+    # 
+    existing_mappings=[(value['value'],value_id_to_concept_map[str(value['id'])],value['id']) for value in value_names]
+    
+    value_name_to_id_map={}
+    for name in names_list:
+        mappings_matching_value_name = [mapping for mapping in existing_mappings if mapping[0] == name]
```

Comment:
In the values case, we want to not only match the name of the value, but also the name of the field it is within, and the value description (if provided). I fear this will require a few more lines to get all that information into this one place.

Reply:
Implemented and seems to work but did it quite fast. Will need to double check that everything is correct when i'm back on thursday